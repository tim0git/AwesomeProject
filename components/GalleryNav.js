import React from 'react';
import {createStackNavigator} from '@react-navigation/stack';
import Gallery from './gallery/Gallery';
import Forrest from './gallery/Forrest';
import Beach from './gallery/Beach';
import Mountains from './gallery/Mountains';
import City from './gallery/City';

const Stack = createStackNavigator();

function GalleryNav(props) {
  return (
    <Stack.Navigator initialRouteName={Gallery}>
      <Stack.Screen name="Gallery" component={Gallery} />
      <Stack.Screen name="Forrest" component={Forrest} />
      <Stack.Screen name="Beach" component={Beach} />
      <Stack.Screen name="City" component={City} />
      <Stack.Screen name="Mountains" component={Mountains} />
    </Stack.Navigator>
  );
}

export default GalleryNav;
