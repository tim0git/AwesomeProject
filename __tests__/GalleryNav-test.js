import 'react-native';
import React from 'react';
import 'jest-enzyme';
import 'react-native-mock-render/mock';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import GalleryNav from '../components/GalleryNav';

// import App from '../App';
// import Hello from '../components/Hello';
// import Details from '../components/Details';

describe('<GalleryNav >', () => {
  const galleryNav = <GalleryNav />;

  it('Component Renders', () => {
    const component = shallow(galleryNav);
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('has a stack navigator', () => {
    const wrapper = shallow(galleryNav);
    expect(wrapper.find('StackNavigator').at(0)).toExist();
  });

  it('stack navigator has 5 navigation screens', () => {
    const count = [0, 1, 2, 3, 4];
    const wrapper = shallow(galleryNav);
    count.forEach((number) => {
      expect(wrapper.find('Screen').at(number)).toExist();
    });
  });

  it('stack navigator has only 5 navigation screens', () => {
    const wrapper = shallow(galleryNav);
    expect(wrapper.find('Screen').at(5).exists()).toBe(false);
  });

  it('screen 1 has text Gallery', () => {
    const wrapper = shallow(galleryNav);
    expect(wrapper.find('Screen').first().props().name).toBe('Gallery');
  });

  it('screen 2 has text Forrest', () => {
    const wrapper = shallow(galleryNav);
    expect(wrapper.find('Screen').at(1).props().name).toBe('Mountains');
  });

  it('screen 3 has text Beach', () => {
    const wrapper = shallow(galleryNav);
    expect(wrapper.find('Screen').at(2).props().name).toBe('Forrest');
  });

  it('screen 4 has text City', () => {
    const wrapper = shallow(galleryNav);
    expect(wrapper.find('Screen').at(3).props().name).toBe('Beach');
  });

  it('screen 5 has text Mountains', () => {
    const wrapper = shallow(galleryNav);
    expect(wrapper.find('Screen').last().props().name).toBe('City');
  });

  test.todo('should render a <Forrest > component when you press Forrest');
  test.todo('should render a <Beach > component when you press Beach');
  test.todo('should render a <City > component when you press City');
  test.todo('should render a <Mountains > component when you press Mountains');
});
