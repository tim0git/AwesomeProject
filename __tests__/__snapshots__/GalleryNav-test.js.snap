// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App > Component Renders 1`] = `
<StackNavigator
  initialRouteName={[Function]}
>
  <Screen
    component={[Function]}
    name="Gallery"
  />
  <Screen
    component={[Function]}
    name="Forrest"
  />
  <Screen
    component={[Function]}
    name="Beach"
  />
  <Screen
    component={[Function]}
    name="City"
  />
  <Screen
    component={[Function]}
    name="Mountains"
  />
</StackNavigator>
`;
